src/oas2/transformers/__tests__/request.test.ts

```diff
@@ -7,81 +7,170 @@ import {
+          schema: {
+            'x-stoplight-id': expect.any(String),
+          },
+        },
+      ],
+    });
   });
 
   it('given single query param should translate to request with query', () => {
-    expect(translateToRequest({}, [fakeQueryParameter])).toMatchSnapshot();
+    expect(translateToRequest([fakeQueryParameter])).toMatchSnapshot({
+      query: [
+        {
+          id: expect.any(String),
+          schema: {
+            'x-stoplight-id': expect.any(String),
+          },
+        },
+      ],
+    });
   });
 
   it('given single header param should translate to request with header', () => {
-    expect(translateToRequest({}, [fakeHeaderParameter])).toMatchSnapshot();
+    expect(translateToRequest([fakeHeaderParameter])).toMatchSnapshot({
+      headers: [
+        {
+          id: expect.any(String),
+          schema: {
+            'x-stoplight-id': expect.any(String),
+          },
+        },
+      ],
+    });
   });
 
   it('given two query params should translate', () => {
-    expect(translateToRequest({}, [fakeQueryParameter, fakeQueryParameter])).toMatchSnapshot();
+    expect(translateToRequest([fakeQueryParameter, fakeQueryParameter])).toMatchSnapshot({
+      query: [
+        {
+          id: expect.any(String),
+          schema: {
+            'x-stoplight-id': expect.any(String),
+          },
+        },
+      ],
+    });
   });
 
   it('given two header params should translate', () => {
-    expect(translateToRequest({}, [fakeHeaderParameter, fakeHeaderParameter])).toMatchSnapshot();
+    expect(translateToRequest([fakeHeaderParameter, fakeHeaderParameter])).toMatchSnapshot({
+      headers: [
+        {
+          id: expect.any(String),
+          schema: {
+            'x-stoplight-id': expect.any(String),
+          },
+        },
+      ],
+    });
   });
 
   it('given two path params should translate', () => {
-    expect(translateToRequest({}, [fakePathParameter, fakePathParameter])).toMatchSnapshot();
+    expect(translateToRequest([fakePathParameter, fakePathParameter])).toMatchSnapshot({
+      path: [
+        {
+          id: expect.any(String),
+          schema: {
+            'x-stoplight-id': expect.any(String),
+          },
+        },
+      ],
+    });
   });
 
   it('should translate mixed request', () => {
     expect(
-      translateToRequest({}, [fakeParameter, fakeParameter, fakeParameter, fakeParameter, fakeParameter]),
-    ).toMatchSnapshot();
+      translateToRequest([
+        fakeBodyParameter,
+        fakeQueryParameter,
+        fakePathParameter,
+        fakeHeaderParameter,
+        fakeFormParameter,
+      ]),
+    ).toMatchSnapshot({
+      body: {
+        id: expect.any(String),
+        contents: [
+          {
+            id: expect.any(String),
+          },
+        ],
+      },
+      headers: [
+        {
+          id: expect.any(String),
+          schema: {
+            'x-stoplight-id': expect.any(String),
+          },
```

Comment:
Hmm, is there value in having the empty schema on all the params? Does empty schema basically === no schema?

Reply:
Yeah, true. I didn't touch that logic, but I agree the way it works is suboptimal.